src/Microsoft.Azure.WebJobs.Host/WebJobsEnvironmentVariablesConfigurationSource.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.EnvironmentVariables;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    /// <summary>
+    /// This source exists to replace the default <see cref="EnvironmentVariablesConfigurationSource"/> to allow
+    /// us to override caching behavior.
+    /// </summary>
+    internal class WebJobsEnvironmentVariablesConfigurationSource : IConfigurationSource
+    {
+        public IConfigurationProvider Build(IConfigurationBuilder builder)
+        {
+            return new WebHostEnvironmentVariablesProvider();
+        }
+
+        private class WebHostEnvironmentVariablesProvider : EnvironmentVariablesConfigurationProvider
+        {
+            public WebHostEnvironmentVariablesProvider() : base()
+            {
+            }
+
+            public override bool TryGet(string key, out string value)
+            {
+                value = Environment.GetEnvironmentVariable(key);
```

Comment:
It looks like the EnvConfigProvider does some normalization that we'd be bypassing here: https://github.com/aspnet/Configuration/blob/dev/src/Config.EnvironmentVariables/EnvironmentVariablesConfigurationProvider.cs.

Reply:
I say we skip that funky logic - for environment variables we just return the truth. I'm not sure what that logic is for, but we've never had it for Functions and we've been fine :)

Reply:
The "__" -> ":" for hierarchical settings was what I was most worried about. From [the docs](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-2.1&tabs=basicconfiguration#configuration-by-environment):

`For hierarchical config values specified in environment variables, a colon (:) may not work on all platforms. Double underscore (__) is supported by all platforms.`

Reply:
Updated the code to support this normalization, and added test verifying that and case insensitivity as well